store/json.go

```diff
@@ -55,9 +57,115 @@ func NewJsonFileStore(fileName string) (KeyValueStore, error) {
 	}
 
 	kvs := &jsonFileStore{
-		fileName: fileName,
+		fileName:     fileName,
 		lockFileName: platform.CNILockPath + filepath.Base(fileName) + lockExtension,
-		data:     make(map[string]*json.RawMessage),
+		data:         make(map[string]*json.RawMessage),
+	}
+
+	return kvs, nil
+}
+
+// fileExists states the passed filename and returns if it exists.
+// if any error except for os.ErrNotExists is encountered, it is returned.
+func fileExists(filename string) (bool, error) {
+	if _, err := os.Stat(filename); err != nil {
+		if errors.Is(err, os.ErrNotExist) {
```

Comment:
can probably use `os.IsNotExist(err)` in lieu of manually checking error type

Reply:
offline, apparently os.IsNotExist(err) is deprecated

Reply:
full context: errors.Is/As should be preferred to the built-in error type functions because they correctly handle wrapped errors and legacy built-ins like os.IsNotExist are not guaranteed to